src/components/fields/MultiTagArrayField.js

```diff
@@ -0,0 +1,163 @@
+					<Affix getContainer={this.getContainerRef}
+						   topOffset={this.props.formContext.topOffset + 15}
+						   bottomOffset={this.props.formContext.bottomOffset}>
+						<div className={"btn-group-vertical"}>
+							{ buttons.map((btnProps, idx) => (
+								<Button
+									key={idx}
+									onClick={this.onButtonClick(idx, btnProps)}
+									variant={btnProps.operation === "delete" ? "outline-danger": "default"}
+									className={classNames(btnProps.className, activeButtonIdx === idx ? "active" : "")}
+								>{ btnProps.label }</Button>
+							)) }
+						</div>
+					</Affix>
+				</Col>
+				<Col xs={9} sm={9} md={10} lg={10} className={"laji-form-multi-tag-array-field-content"}>
+					{ propertyKeys.map(key => (
+						<SchemaField
+							{...this.props}
+							key={key}
+							schema={{title: "", ...schema.properties[key]}}
+							uiSchema={{
+								"ui:field": "TagArrayField",
+								"ui:options": {
+									...uiOptions,
+									buttons: undefined,
+									onTagClick: this.onTagClick(key)
+								}
+							}}
+							errorSchema={errorSchema[key] || {}}
+							idSchema={idSchema[key]}
+							formData={formData[key] || []}
+							required={uiSchema[key]?.["ui:required"] || false}
+							onChange={this.onChange(key)}
+						/>
+					)) }
+				</Col>
+			</Row>
+		);
+	}
+
+	onChange = (key) => (formData) => {
+		const newFormData = {...this.props.formData, [key]: formData};
+		this.props.onChange(newFormData);
+	}
+
+	onButtonClick = (idx, {operation, fromField, toField}) => () => {
+		const {activeButtonIdx} = this.state;
+		if (activeButtonIdx === idx) {
+			this.setState({activeButtonIdx: undefined});
+			return;
+		}
+
+		if (operation === "moveAll") {
+			const toFieldNewValue = (this.props.formData[toField] || []).concat(this.props.formData[fromField] || []);
+			const newFormData = {...this.props.formData, [fromField]: [], [toField]: toFieldNewValue};
+			this.props.onChange(newFormData);
+		} else if (operation === "move" || operation === "delete") {
+			this.setState({activeButtonIdx: idx});
+		}
+	}
+
+	onTagClick = (fromField) => (idx) => {
+		const {activeButtonIdx} = this.state;
+		if (activeButtonIdx === undefined) {
+			return;
+		}
+
+		const {buttons = []} = getUiOptions(this.props.uiSchema);
+		const {operation, toField} = buttons[activeButtonIdx];
+		if (fromField === toField) {
+			return;
+		}
+
+		let formData = this.props.formData;
+
+		if (operation === "move" || operation === "delete") {
+			const value = formData[fromField][idx];
+			const fromFieldNewValue = [...formData[fromField]];
+			fromFieldNewValue.splice(idx, 1);
+
+			if (operation === "move") {
+				const toFieldNewValue = [...(formData[toField] || []), value]
+				formData = {...formData, [fromField]: fromFieldNewValue, [toField]: toFieldNewValue};
+			} else {
+				formData = {...formData, [fromField]: fromFieldNewValue};
+			}
+
+			this.props.onChange(formData);
+		}
+	}
```

Comment:
We shouldn't use curried functions for JSX props, as they'll always create a new function instance hence triggering false change detection. Use memoization for creating the event handler for each "key". Look at [`SingleActiveArrayField`](https://github.com/luomus/laji-form/blob/2a9426ad17928853b5bcdc9d474e7a59125461d2/src/components/fields/SingleActiveArrayField.js#L837-L855) for an example.